guidance/include/guidance/guidance_state_machine.hpp

```diff
@@ -71,15 +71,57 @@ namespace guidance
              */
             uint8_t getCurrentState();
 
+            virtual void StartUpState(Signal signal) = 0;
+            virtual void DriversReadyState(Signal signal) = 0;
+            virtual void ActiveState(Signal signal) = 0;
+            virtual void EngagedState(Signal signal) = 0;
+            virtual void InactiveState(Signal signal) = 0;
+            virtual void OffState(Signal signal) = 0;
+            // a local variable keeps the current state machine state
+            State current_guidance_state;
+
         private:
             /*!
              * \brief Actual state machine logic driven by signal enum
              */
             void onGuidanceSignal(Signal signal);
 
-            // a local variable keeps the current state machine state
-            State currentGuidanceState;
+    };
+
+    class Cadilac : public GuidanceStateMachine {
+        public:
+            void StartUpState(Signal signal);
+            void DriversReadyState(Signal signal);
+            void ActiveState(Signal signal);
+            void EngagedState(Signal signal);
+            void InactiveState(Signal signal);
+            void OffState(Signal signal);
+    };
 
+    class Lexus : public GuidanceStateMachine {
+        public:
+            void StartUpState(Signal signal);
+            void DriversReadyState(Signal signal);
+            void ActiveState(Signal signal);
+            void EngagedState(Signal signal);
+            void InactiveState(Signal signal);
+            void OffState(Signal signal);
+    };
+
+    class Factory {
```

Comment:
This class name is too generic. I'm also not quite sure what the point of having this factory interface is when there is only a single type of GuidanceStateMachineFactory. It is a nice abstraction but it might be overkill in this case. How likely is it that we have different kinds of factories here.